### PR TITLE
#14: Add support for .NET 4.6.1+ (WIP)

### PR DESCRIPTION
**Pull request by mtkennerly**
`dotnet build` is working (with 10 warnings), but `dotnet test` is failing:

```
Failed!  - Failed:    22, Passed:    56, Skipped:     0, Total:    78, Duration:  - Linguini.Syntax.Tests.dll (net461)
Failed!  - Failed:    52, Passed:  3932, Skipped:     0, Total:  3984, Duration:  - Linguini.Bundle.Test.dll (net461)
```

I'm not sure yet if I messed something up or if it's an issue with the polyfills. I'll try to do some more digging, but I figured I would go ahead and open this as a draft to get your thoughts.

### CODE REVIEW

**Comment by Ygg01** [comment]
Ok off the bat, I would aim to support `net4.6.1` as an additional target, using conditional references to work with 4.6.1

As for the errors, your re-implementation of Syntax method doesn't work. If syntax is wrong, so is everything else.

File: Linguini.Syntax/IO/ZeroCopyReader.cs [hunk 1]
```diff
@@ -160,7 +161,8 @@ public ReadOnlyMemory<char> ReadSlice(int start, int end)
 
         public string ReadSliceToStr(int start, int end)
         {
-            return new(_unconsumedData.Slice(start, end - start).Span);
+            MemoryMarshal.TryGetString(_unconsumedData.Slice(start, end - start), out var text, out var _, out var _);
```
Comment (Ygg01) [inline Linguini.Syntax/IO/ZeroCopyReader.cs:164]: This is incorrect. You want the `var text = MemoryMarshal.TryGetString` .
E.g. I have following string `abc`. I say `ReadSliceToStr(0,1)` and expect `a` the method as written returns `abc`.

I'd also put this into separate `#if net461 block` and keep original imlementation for brevity.

File: Linguini.Bundle.Test/Linguini.Bundle.Test.csproj [hunk 2]
```diff
@@ -24,4 +24,10 @@
         <AdditionalFiles Include="..\PluralRules.Generator\cldr_data\plurals.xml" />
     </ItemGroup>
 
+    <ItemGroup>
+        <PackageReference Include="IndexRange" Version="1.0.2" PrivateAssets="all" />
```
Comment (Ygg01) [inline Linguini.Bundle.Test/Linguini.Bundle.Test.csproj:28]: Honestly I'd use [conditional package references](https://markheath.net/post/csproj-conditional-references)

I.e. 
```xml
<PackageReference Condition=" '$(TargetFramework)' == 'net462' "Include="IndexRange" Version="1.0.2" PrivateAssets="all"  />
```

Or putting it on whole ItemGroup

**Comment by mtkennerly** [comment]
Closing in favor of https://github.com/Ygg01/Linguini/pull/16.

**Pull request closed without merging**